### PR TITLE
D8CORE-2489 - add `aria-current`

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- For the sidebar navigation on pages like /news and /events, we need `aria-current="true"` added to the link of the page the user is currently on.

# Review By (Date)
- 10/6/21

# Criticality
- 3

# Urgency
- normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. clear drupal caches
3. visit `/news/awards`. Verify the link in the sidebar nav labeled "Awards" has the `aria-current` attribute set to "true".  Note, that should be on the link, not the <li>.
4. visit `/events/seminar` and verify the Seminar link also has the `aria-current` attribute.
5. Same thing should be true for `/people` and `/publications` as well.

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? NO

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: stanford_basic.theme [hunk 1]
```diff
@@ -226,12 +226,26 @@ function stanford_basic_preprocess_block__system_branding_block(&$vars) {
  * Implements hook_preprocess_menu().
  */
 function stanford_basic_preprocess_menu(&$variables, $hook) {
+
+  // This should run for every menu.
   $current_path = \Drupal::request()->getRequestUri();
+  $items = $variables['items'];
+  foreach ($items as $key => $item) {
+    // If path is current_path, add aria-current to the link.
+    if ($item['url']->toString() == $current_path) {
```
Comment (pookmish) [inline stanford_basic.theme:235]: This doesn't check for arguments or anchor links as part of the link.
```suggestion
    if (parse_url($item['url']->toString(), PHP_URL_PATH) == $current_path) {
```

**Review by pookmish — CHANGES_REQUESTED** [review]
Just one change request

File: stanford_basic.theme (hunk 1 above)
Comment (imonroe) [reply stanford_basic.theme]: That's a good suggestion, thanks.

**Review by pookmish — APPROVED** [review]

**Comment by imonroe** [comment]
@cjwest I couldn't run the HTML validation until the code was up somewhere public.  The HTML validators don't work on localdev.  It's kind of a chicken-and-egg thing; I can't run online tests until the code is somewhere public, it won't be anywhere public until it's deployed, it can't be deployed until it's merged, and the checkboxes want me to check them before it's merged.

**Pull request merged**